pulsar-broker/src/main/java/org/apache/pulsar/broker/service/PulsarCommandSenderImpl.java

```diff
@@ -243,8 +243,12 @@ public ChannelPromise sendMessagesToConsumer(long consumerId, String topicName,
                 // increment ref-count of data and release at the end of process:
                 // so, we can get chance to call entry.release
                 metadataAndPayload.retain();
-                // skip raw message metadata since broker timestamp only used in broker side
-                Commands.skipBrokerEntryMetadataIfExist(metadataAndPayload);
+                // skip broker entry metadata if consumer-client doesn't support broker entry metadata or the
+                // features is not enabled
+                if (cnx.getRemoteEndpointProtocolVersion() < ProtocolVersion.v18.getValue()
+                        || !cnx.isExposingBrokerEntryMetadataToClientEnabled()) {
```

Comment:
We can get the configuration through the following dependency paths
PulsarCommandSenderImpl -> ServerCnx -> BrokerService -> PulsarService -> configuration
`isExposingBrokerEntryMetadataToClientEnabled` in ServerCnx can be removed

Reply:
Thanks for your suggestion. I've removed it in ServerCnx